bgpd/bgp_evpn.c

```diff
@@ -2562,6 +2636,76 @@ int bgp_evpn_unimport_route(struct bgp *bgp, afi_t afi, safi_t safi,
 	return install_uninstall_evpn_route(bgp, afi, safi, p, ri, 0);
 }
 
+/* filter routes which have martian next hops */
+int bgp_filter_evpn_routes_upon_martian_nh_change(struct bgp *bgp)
+{
+	afi_t afi;
+	safi_t safi;
+	struct bgp_node *rd_rn, *rn;
+	struct bgp_table *table;
+	struct bgp_info *ri;
+
+	afi = AFI_L2VPN;
+	safi = SAFI_EVPN;
+
+	/* Walk entire global routing table and evaluate routes which could be
+	 * imported into this VPN. Note that we cannot just look at the routes
+	 * for the VNI's RD -
+	 * remote routes applicable for this VNI could have any RD.
+	 */
+	/* EVPN routes are a 2-level table. */
+	for (rd_rn = bgp_table_top(bgp->rib[afi][safi]); rd_rn;
+	     rd_rn = bgp_route_next(rd_rn)) {
+		table = (struct bgp_table *)(rd_rn->info);
+		if (!table)
+			continue;
+
+		for (rn = bgp_table_top(table); rn; rn = bgp_route_next(rn)) {
+
+			for (ri = rn->info; ri; ri = ri->next) {
+
+				/* Consider "valid" remote routes applicable for
+				 * this VNI. */
+				if (!(ri->type == ZEBRA_ROUTE_BGP
+				      && ri->sub_type == BGP_ROUTE_NORMAL))
+					continue;
+
+				if (bgp_nexthop_self(bgp,
+					ri->attr->nexthop)) {
+
+					char attr_str[BUFSIZ];
+					char pbuf[PREFIX_STRLEN];
+
+					bgp_dump_attr(ri->attr, attr_str,
+						      BUFSIZ);
+
+					if (bgp_debug_update(ri->peer, &rn->p,
+							     NULL, 1))
+						zlog_debug(
```

Comment:
debug statement needs to be under a condition that ties it to the corresponding `debug bgp ...` config to prevent logspam